internal/stacks/stack_manager.go

```diff
@@ -430,8 +439,8 @@ func checkPortAvailable(port int) (bool, error) {
 func (s *StackManager) runFirstTimeSetup(verbose bool, options *StartOptions) error {
 	workingDir := filepath.Join(constants.StacksDir, s.Stack.Name)
 
-	s.Log.Info("initializing blockchain node")
-	if err := s.blockchainProvider.RunFirstTimeSetup(); err != nil {
+	s.Log.Info("performing early initialization tasks")
+	if err := s.blockchainProvider.PreFirstTimeSetup(); err != nil {
```

Comment:
I think both this and the tokens provider should have this function named the same thing. It feels strange to have a `PreFirstTimeSetup` without the `PostFirstTimeSetup` or vice versa.

Reply:
Yea, I was drifting toward defining a common set of lifecycle hooks and letting all providers implement the ones they need - but that may be premature. I'll name them both `FirstTimeSetup` for now, even though they're invoked from different points. We can continue to revisit as the "providers" concept evolves.